src/pages/GamePage/GameWord/GameWordContainer.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useRef } from 'react';
+import { useForm } from 'react-hook-form';
+import Dashboard from '@/common/Ingame/Dashboard';
+import useTypingState from '../GameSentence/useTypingState';
+import { InagmeWsChildrenProps } from '../IngameWSErrorBoundary';
+import WordCell from './WordCell';
+import WordRankContainer from './WordRankContainer';
+
+interface GameWordProps extends InagmeWsChildrenProps {
+  userId: number;
+}
+
+const GameWordContainer = ({
+  ingameRoomRes,
+  publishIngame,
+  userId,
+}: GameWordProps) => {
+  const { register, handleSubmit, setValue, getValues } = useForm();
+  const { cpm, onInputChange, initializeTyping } = useTypingState();
+
+  const submitCount = useRef(0);
+  const approvedSubmitCount = useRef(0);
```

Comment:
approvedSubmitCount는 ingameRoomRe.allMembers.score와 같은 값으로 추측되는데, 서버에서 상태인 score를 쓰시는건 어떠신가용?!

Reply:
```suggestion
  const approvedSubmitCountFromScore =
    ingameRoomRes.allMembers.find(({ memberId }) => memberId === userId)?.score ?? 0;
```
저번에 슬랙에서 얘기한거 처럼 이렇게 계산으로 두면, 타이핑입력으로 cpm 이 바뀔때마다 이연산을 매번 실행하는것 같아서요!
ref에 두고 입력성공 응답이 왔는데 본인일때만 하나씩 증가하면 연산이 조금 덜하지 않을까? 생각했습니다.
차이가 별로 없을까요?

Reply:
저도 영현님 방식으로 해보니까 allMembers나 find에서 undefined가 넘어올 때가 있어서 아직은 useRef를 사용하고 있는데, 영현님 방어코드를 어떻게 넣으셨는지 궁금하네요 ㅎㅎ 🤔